### PR TITLE
Update task summary part 1

### PR DESCRIPTION
**Pull request by stevhliu**
This PR reworks the task summary to be more conceptual and provides more explanation about a topic to help users better understand it. It'll be focused more on understanding instead of practical steps. The update will be split into two parts:

1. Describe the tasks 🤗 Transformers is capable of solving (the focus of this PR). Provide some context about how these tasks used to be solved, how they're handled now, and practical applications of each task.
2. Explain how 🤗 Transformers solve these tasks. This'll be a more conceptually advanced and separate page.

### CODE REVIEW

File: docs/source/en/task_summary.mdx [hunk 1]
```diff
@@ -10,1125 +10,222 @@ an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express o
...
->>> print(model.config.id2label[predicted_label])
-Egyptian cat
+>>> text = "Hugging Face is a <mask> company based in New York City."
+>>> fill_mask = pipeline(task="fill-mask")
+>>> fill_mask(text, top_k=3)
 ```
-</pt>
-</frameworkcontent>
+
+Hopefully, this has given you some more background information about all these tasks and the practical importance of each one. The Transformer really changed the paradigm for how we approach all these different tasks and it's exciting to see how Transformer models are being used to solve problems new and old. In the next section, you'll learn **how** 🤗 Transformers work to solve these tasks.
```
Comment (MKhalusova) [inline docs/source/en/task_summary.mdx:231]: Will the "next section" be part of this document or do you plan to have a separate doc for the "how"?

**Review by MKhalusova — COMMENTED** [review]
Very well-written doc, and a pleasure to read. Excellent work! I feel like it could use some "learn more" links per task, or some other way of navigating from the conceptual description of a task to a deeper exploration of the topic. Though I think you had something along these lines planned for future iterations.

File: docs/source/en/task_summary.mdx [hunk 2]
```diff
@@ -10,1125 +10,222 @@ an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express o
...
-...     f"Answer: '{result['answer']}', score: {round(result['score'], 4)}, start: {result['start']}, end: {result['end']}"
-... )
-Answer: 'the task of extracting an answer from a text given a question', score: 0.6177, start: 34, end: 95
-
->>> result = question_answerer(question="What is a good example of a question answering dataset?", context=context)
->>> print(
-...     f"Answer: '{result['answer']}', score: {round(result['score'], 4)}, start: {result['start']}, end: {result['end']}"
-... )
-Answer: 'SQuAD dataset', score: 0.5152, start: 147, end: 160
+>>> transcriber = pipeline(task="automatic-speech-recognition")
```
[the remark this answers is not in the document]
Comment (stevhliu) [reply docs/source/en/task_summary.mdx:52]: Good question! I don't think there is an argument in the pipeline for setting a different language. You would need to use a multilingual model or one finetuned on a specific language. At the moment, I can't remember what the default model is for ASR. 🫣

We briefly address your question [here](https://huggingface.co/docs/transformers/main/en/quicktour#use-another-model-and-tokenizer-in-the-pipeline), but we don't explicitly say it. Maybe it can be clarified there or in the [pipeline tutorial](https://huggingface.co/docs/transformers/main/en/pipeline_tutorial)?

File: docs/source/en/task_summary.mdx [hunk 3]
```diff
@@ -10,1125 +10,222 @@ an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express o
...
-
-This outputs a (hopefully) coherent next token following the original sequence, which in our case is the word *is* or
-*features*.
 
-In the next section, we show how [`generation.GenerationMixin.generate`] can be used to
-generate multiple tokens up to a specified length instead of one token at a time.
+### Object detection
 
-### Text Generation
+Object detection identifies multiple objects within an image unlike image classification. This gives you more information about the different instances of objects in an image as well as the positions of the objects in an image (defined by the bounding box). Some example applications of object detection include:
```
[the remark this answers is not in the document]
Comment (stevhliu) [reply docs/source/en/task_summary.mdx:82]: Much more concise, thank you!

File: docs/source/en/task_summary.mdx (hunk 1 above)
Comment (stevhliu) [reply docs/source/en/task_summary.mdx:231]: It'll be a separate [doc](https://github.com/huggingface/transformers/pull/21067) for the "how".

File: docs/source/en/task_summary.mdx [hunk 4]
```diff
@@ -10,1125 +10,224 @@ an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express o
 specific language governing permissions and limitations under the License.
 -->
 
-# Summary of the tasks
+# What 🤗 Transformers can do
 
-[[open-in-colab]]
+The [Transformer](https://arxiv.org/abs/1706.03762) is a uniquely flexible and powerful architecture. While it is typically associated with natural language processing (NLP) tasks, other fields like computer vision and speech processing are increasingly adopting Transformer architectures to solve tasks in their area. If you look at some of the underlying technology behind popular consumer products like smartphones, apps, and self-driving cars, odds are that you'll find some variant of a Transformer behind it. That is because the Transformer is a general architecture that can be applied to many tasks!
```
Comment (sgugger) [inline docs/source/en/task_summary.mdx:15]: I wouldn't frame this around the Transformer model as we have a few models that are non-Transformer in the library and it's not clear that Transformer architectures are better than convnets in computer vision. If a non-Transformer architecture gets state-of-the-art on any task, we will include it.

So I would remove this paragraph.

File: docs/source/en/task_summary.mdx [hunk 5]
```diff
@@ -10,1125 +10,224 @@ an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express o
...
 
-These examples leverage auto-models, which are classes that will instantiate a model according to a given checkpoint,
-automatically selecting the correct model architecture. Please check the [`AutoModel`] documentation
-for more information. Feel free to modify the code to be more specific and adapt it to your specific use-case.
+## Audio
 
-In order for a model to perform well on a task, it must be loaded from a checkpoint corresponding to that task. These
-checkpoints are usually pre-trained on a large corpus of data and fine-tuned on a specific task. This means the
-following:
+Classical speech and audio processing techniques typically involved Gaussian mixture models, Hidden Markov models, and recurrent neural networks (RNNs). But these have been eclipsed by more modern approaches like [sequence-to-sequence models](./glossary#sequencetosequence-seq2seq) and [connectionist temporal classification (CTC)](./glossary#connectionist-temporal-classification-ctc), which dramatically simplify training end-to-end audio and speech processing systems and improve performance.
```
Comment (sgugger) [inline docs/source/en/task_summary.mdx:21]: I don't think the user cares about this at all for this introductory guide.

File: docs/source/en/task_summary.mdx [hunk 6]
```diff
@@ -10,1125 +10,224 @@ an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express o
...
 
-- Not all models were fine-tuned on all tasks. If you want to fine-tune a model on a specific task, you can leverage
-  one of the *run_$TASK.py* scripts in the [examples](https://github.com/huggingface/transformers/tree/main/examples) directory.
-- Fine-tuned models were fine-tuned on a specific dataset. This dataset may or may not overlap with your use-case and
-  domain. As mentioned previously, you may leverage the [examples](https://github.com/huggingface/transformers/tree/main/examples) scripts to fine-tune your model, or you may
-  create your own training script.
+The [wav2vec2](./model_doc/wav2vec2) model was one of the first successful Transformer-based speech processing models. It adapted some of the ideas that made pretrained language models like BERT so successful. For instance, wav2vec2 pretrained on raw unlabeled speech and used a Transformer encoder to contextualize the speech representations. Its pretraining objective was to predict masked parts of the audio, just like the [masked language modeling](./glossary#masked-language-modeling) objective in BERT!
```
Comment (sgugger) [inline docs/source/en/task_summary.mdx:23]: Same here, a user doesn't care that Wav2Vec2 is the same as BERT, they just want to know how to do a task. Wav2Vec2 might not even be the best model for these tasks in a couple of months.

File: docs/source/en/task_summary.mdx [hunk 7]
```diff
@@ -10,1125 +10,224 @@ an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express o
...
-Answer: general - purpose architectures
-Question: 🤗 Transformers provides interoperability between which frameworks?
-Answer: tensorflow 2. 0 and pytorch
-```
-</tf>
-</frameworkcontent>
+## Computer vision
 
-## Language Modeling
+One of the first and earliest successful computer vision applications was recognizing zip code numbers using a [convolutional neural network (CNN)](glossary#convolution). For many years, CNNs have been the go-to for computer vision tasks, and modern CNNs like [ConvNeXT](./model_doc/convnext) continue to see strong usage with competitive performance.
```
Comment (sgugger) [inline docs/source/en/task_summary.mdx:58]: Same here. This is the summary of tasks, not the summary of models. I don't think the user comes to this guide to get a history lesson/comparative guide on architectures.

File: docs/source/en/task_summary.mdx [hunk 8]
```diff
@@ -10,1125 +10,224 @@ an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express o
...
-   model, such as `Bart` or `T5`.
-2. Define the article that should be summarized.
-3. Add the T5 specific prefix "summarize: ".
-4. Use the `PreTrainedModel.generate()` method to generate the summary.
+## Natural language processing
 
-In this example we use Google's T5 model. Even though it was pre-trained only on a multi-task mixed dataset (including
-CNN / Daily Mail), it yields very good results.
+The Transformer was introduced in the, [*Attention Is All You Need*](https://arxiv.org/abs/1706.03762), paper for tasks such as language modeling, translation, and summarization. At the time, NLP tasks were typically solved with long short-term memory (LSTM) networks and recurrent neural networks (RNNs). But the Transformer introduced a new architecture based entirely on self-attention mechanisms. These attention layers allow the Transformer to focus on specific words when learning the representation of each word. As a result, the Transformer didn't only outperform previous models, but it was also more efficient and faster to train! Since then, Transformer models have driven most of the recent progress in NLP.
```
Comment (sgugger) [inline docs/source/en/task_summary.mdx:129]: Same here

File: docs/source/en/task_summary.mdx [hunk 9]
```diff
@@ -10,1125 +10,224 @@ an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express o
...
 
-<frameworkcontent>
-<pt>
-```py
->>> from transformers import AutoModelForSeq2SeqLM, AutoTokenizer
-
->>> model = AutoModelForSeq2SeqLM.from_pretrained("t5-base")
->>> tokenizer = AutoTokenizer.from_pretrained("t5-base")
+* extractive: given a question and some context, the answer is a span of text from the context the model must extract
+* abstractive: given a question and some context, the answer is generated from the context
```
Comment (sgugger) [inline docs/source/en/task_summary.mdx:164]: This one is not dealt with by the question-answering pipeline but by the seq2seq pipeline. This should be mentioned to the user.

File: docs/source/en/task_summary.mdx [hunk 10]
```diff
@@ -10,1125 +10,224 @@ an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express o
...
 
-Audio classification assigns a class to an audio signal. The Keyword Spotting dataset from the [SUPERB](https://huggingface.co/datasets/superb) benchmark is an example dataset that can be used for audio classification fine-tuning. This dataset contains ten classes of keywords for classification. If you'd like to fine-tune a model for audio classification, take a look at the [run_audio_classification.py](https://github.com/huggingface/transformers/blob/main/examples/pytorch/audio-classification/run_audio_classification.py) script or this [how-to guide](./tasks/audio_classification).
+Like question answering, there are two types of summarization:
 
-The following examples demonstrate how to use a [`pipeline`] and a model and tokenizer for audio classification inference:
+* extractive: identify and extract the most important sentences from the original text
```
Comment (sgugger) [inline docs/source/en/task_summary.mdx:184]: This is not what the summarization pipeline does.

**Review by sgugger — COMMENTED** [review]
Thanks for working on the rewrite of this guide. It is called "what Transformers can do", so I think the introductions of each modality should just explain what each modality is and abstain from comparing models, or even mention model names. None of the examples actually show which model is being used, so the user definitely doesn't need that info.

File: docs/source/en/task_summary.mdx (hunk 4 above)
Comment (stevhliu) [reply docs/source/en/task_summary.mdx:15]: I reframed this around the 🤗 Transformers library and what it can do instead of the Transformer model.

**Comment by stevhliu** [comment]
> It is called "what Transformers can do", so I think the introductions of each modality should just explain what each modality is and abstain from comparing models

Thanks for the feedback, this helped me refine the scope of this page! It was a little difficult trying to discuss the tasks and not be tempted to also talk about the models since the two are so closely related. 😅 I updated the intro of each modality with an explanation of the input data and how to get it into a useable format by the model to solve a task.

**Review by sgugger — APPROVED** [review]
It's way better this way, thanks!

**Review by MKhalusova — APPROVED** [review]
LGTM!
